e2e/test/PnpServiceTests.cs

```diff
@@ -0,0 +1,83 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading.Tasks;
+using FluentAssertions;
+using Microsoft.Azure.Devices.Client;
+using Microsoft.Azure.Devices.Shared;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Microsoft.Azure.Devices.E2ETests
+{
+    /// <summary>
+    /// Test class containing all tests to be run for plug and play.
+    /// </summary>
+    [TestClass]
+    [TestCategory("E2E")]
+    [TestCategory("IoTHub")]
+    [TestCategory("PlugAndPlay")]
+    public class PnpServiceTests
+    {
+        private const string DevicePrefix = "plugAndPlayDevice";
+        private const string ModulePrefix = "plugAndPlayModule";
+        private const string TestModelId = "dtmi:com:example:testModel;1";
+
+        [TestMethod]
+        public async Task DeviceTwin_Contains_ModelId()
+        {
+            // Setup
+
+            // Create a device.
+            TestDevice testDevice = await TestDevice.GetTestDeviceAsync(DevicePrefix).ConfigureAwait(false);
+            // Send model ID with MQTT connect packet to make the device plug and play.
+            var options = new ClientOptions
+            {
+                ModelId = TestModelId,
+            };
+            using var deviceClient = DeviceClient.CreateFromConnectionString(testDevice.ConnectionString, Client.TransportType.Mqtt_Tcp_Only, options);
```

Comment:
hey, I am adding test case for x509 auth as well in addition to this, just talked to Tim about it that we should test that flow as well.

Reply:
Sure. In this specific case it probably makes sense though it would be testing their SDK.